aopy/visualization/base.py

```diff
@@ -2498,4 +2498,115 @@ def plot_plane(plane, ax=None, gain=1.0, color='grey', alpha=0.15, resolution=10
         raise ValueError(f"Invalid mapping shape {plane.shape}. Expected (3,3) or (4,4) \
                          for transformation matrices, or (4,) for plane equations.")
         
-    ax.plot_surface(x, y, z, alpha=alpha, color=color)
\ No newline at end of file
+    ax.plot_surface(x, y, z, alpha=alpha, color=color)
+
+def plot_sphere(location, color='gray', radius=4, resolution=20, alpha=1, ax=None, **kwargs):
+    """
+    Plots a 3D sphere on a specified 3D Matplotlib axis.
+
+    Args:
+        location (tuple or list): Coordinates of the sphere's center, specified as (x, y, z).
+        color (str, optional): Color of the sphere. Default is 'gray'.
+        radius (float, optional): Radius of the sphere. Default is 4.
+        resolution (int, optional): Number of subdivisions for the sphere's surface. Higher values 
+            result in a smoother appearance but may reduce performance. Default is 20.
+        alpha (float, optional): Transparency of the sphere, where 1 is opaque and 0.4 is fully 
+            transparent. Must be between 0.4 and 1. Default is 1.
+        ax (mpl_toolkits.mplot3d.Axes3D, optional): The Matplotlib 3D axis on which to plot the sphere.
+
+    Raises:
+        ValueError: If the 'alpha' value is outside the valid range [0.4, 1].
+
+    Examples:
+        To plot a semi-transparent blue sphere with a radius of 1 at the origin:
+
+        .. code-block:: python
+
+            import matplotlib.pyplot as plt
+            from mpl_toolkits.mplot3d import Axes3D
+
+            fig = plt.figure()
+            ax = fig.add_subplot(111, projection='3d')
+            plot_sphere(location=(0, 1, 2), color='blue', radius=5, resolution=30, alpha=0.5, ax=ax)
+
+        .. image:: _images/plot_sphere_example.png
+    """
+    if ax is None:
+        fig = plt.figure()
+        ax = fig.add_subplot(111, projection='3d')
+    
+    # Assert that alpha is within the valid range [0.4, 1]
+    if not (0.4 <= alpha <= 1):
+        raise ValueError(f"Invalid value for transparency (alpha): {alpha}. Must be between 0.4 and 1.")
+    
+    # Generate points in spherical coordinates
+    phi = np.linspace(0, 2 * np.pi, resolution) # azimuthal angle
+    theta = np.linspace(0, np.pi, resolution) # polar angle
+    
+    # Translate to cartesian coordinates
+    x = radius * np.outer(np.cos(phi), np.sin(theta)) + location[0]
+    y = radius * np.outer(np.sin(phi), np.sin(theta)) + location[1]
+    z = radius * np.outer(np.ones(np.size(phi)), np.cos(theta)) + location[2]
+    
+    # Plot sphere
+    ax.plot_surface(x, y, z, color=color, alpha=alpha-0.4)
+
+def plot_targets_3D(target_locations, colors, target_radius=1, resolution=20, alpha=1, ax=None, **kwargs):
```

Comment:
this function is more similar to the 2d version of color_targets() than it is to plot_targets(), maybe worth changing the name to color_targets_3d?

Reply:
the **kwargs isn't used, you can probably remove it